.github/workflows/test.yml

```diff
@@ -13,12 +13,13 @@ jobs:
           - windows-latest
           - macos-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set up Go
+      - name: Setup Go
         uses: actions/setup-go@v3
         with:
-          go-version: '1.17.0'
+          go-version: '1.17'
+          check-latest: true
```

Comment:
As we chatted about out of band, I've never seen this needed with public runners, as they will tend to correctly pick the latest available minor version pretty much as soon as its released. Otherwise this is something we should apply globally for consistency. I guess I'm really in camp "use the defaults" unless we have an explicit need.

```suggestion
```